src/misc/unicode.cpp

```diff
@@ -1079,19 +1095,18 @@ uint16_t UTF8_GetCodePage()
         return cp_default;
 }
 
-void UTF8_RenderForDos(const std::string &str_in, std::string &str_out,
-                       const uint16_t code_page)
+std::string UTF8_RenderForDos(const std::string &str_in,
+                              const uint16_t code_page)
 {
     load_config_if_needed();
     const uint16_t cp = deduplicate_code_page(code_page);
     prepare_code_page(cp);
 
     std::vector<uint16_t> str_wide;
-    utf8_to_wide(str_in, str_wide);
+    if (!utf8_to_wide(str_in, str_wide))
+        LOG_WARNING("UTF8: Problem rendering string");
```

Comment:
This is a very good example for showing what a calling method may do when the conversion may not be done properly. Thanks for the update!